funding/manager_test.go

```diff
@@ -3520,3 +3520,166 @@ func TestWumboChannelConfig(t *testing.T) {
+			expectErr: true,
+		},
+		{
+			name:      "nonstandard script",
+			pkscript:  []byte("\x99"),
+			expectErr: true,
+		},
+		{
+			name: "p2sh script",
+			pkscript: []byte("\xa9\x14\xfe\x44\x10\x65\xb6\x53" +
+				"\x22\x31\xde\x2f\xac\x56\x31\x52\x20\x5e" +
+				"\xc4\xf5\x9c\x74\x87"),
+			expectErr: false,
+		},
+		{
+			name: "p2pkh script",
+			pkscript: []byte("\x76\xa9\x14\x64\x1a\xd5\x05\x1e" +
+				"\xdd\x97\x02\x9a\x00\x3f\xe9\xef\xb2\x93" +
+				"\x59\xfc\xee\x40\x9d\x88\xac"),
+			expectErr: false,
+		},
+		{
+			name: "p2wpkh script",
+			pkscript: []byte("\x00\x14\x4b\xfe\x98\x3f\x16\xaa" +
+				"\xde\x1b\x1c\xb1\x54\x5a\xa5\xa4\x88\xd5" +
+				"\xe3\x68\xb5\xdc"),
+			expectErr: false,
+		},
+		{
+			name: "p2wsh script",
+			pkscript: []byte("\x00\x20\x1d\xd6\x3c\x20\x13\x89" +
+				"\x3a\x8b\x41\x5e\xb2\xe7\x41\x8f\x07\x5d" +
+				"\x4f\x3b\xf1\x81\x34\x99\xef\x31\xfb\xd7" +
+				"\x8c\xa8\xc4\x5d\x8f\xf0"),
+			expectErr: false,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+			testUpfrontFailure(t, test.pkscript, test.expectErr)
+		})
+	}
+}
+
+func testUpfrontFailure(t *testing.T, pkscript []byte, expectErr bool) {
+	alice, bob := setupFundingManagers(t)
+	defer tearDownFundingManagers(t, alice, bob)
+
+	errChan := make(chan error, 1)
+	updateChan := make(chan *lnrpc.OpenStatusUpdate)
+
+	fundingAmt := btcutil.Amount(500000)
+	pushAmt := lnwire.NewMSatFromSatoshis(btcutil.Amount(0))
+
+	initReq := &InitFundingMsg{
+		Peer:            alice,
+		TargetPubkey:    alice.privKey.PubKey(),
+		ChainHash:       *fundingNetParams.GenesisHash,
+		SubtractFees:    false,
+		LocalFundingAmt: fundingAmt,
+		PushAmt:         pushAmt,
+		FundingFeePerKw: 1000,
+		Private:         false,
+		Updates:         updateChan,
+		Err:             errChan,
+	}
+	bob.fundingMgr.InitFundingWorkflow(initReq)
+
+	// Bob should send an open_channel message to Alice.
+	var bobMsg lnwire.Message
+	select {
+	case bobMsg = <-bob.msgChan:
+	case err := <-initReq.Err:
+		t.Fatalf("received unexpected error: %v", err)
+	case <-time.After(time.Second * 5):
+		t.Fatalf("timed out waiting for bob's message")
+	}
+
+	bobOpenChan, ok := bobMsg.(*lnwire.OpenChannel)
+	if !ok {
+		t.Fatalf("expected to receive OpenChannel, instead received "+
+			"%T", bobMsg)
+	}
+
+	// Set the UpfrontShutdownScript in OpenChannel.
+	bobOpenChan.UpfrontShutdownScript = lnwire.DeliveryAddress(pkscript)
+
+	// Send the OpenChannel message to Alice now. If we expected an error,
+	// check that we received it.
+	alice.fundingMgr.ProcessFundingMsg(bobOpenChan, bob)
+
+	var aliceMsg lnwire.Message
+	select {
+	case aliceMsg = <-alice.msgChan:
+	case <-time.After(time.Second * 5):
+		t.Fatalf("timed out waiting for alice's message")
+	}
+
+	if expectErr {
+		// Assert that aliceMsg is of type lnwire.Error.
+		_, ok = aliceMsg.(*lnwire.Error)
+		if !ok {
+			t.Fatalf("expected to receive Error, instead "+
```

Comment:
nit: could also `require.True` for fewer lines

Reply:
Done